### PR TITLE
Add label selector support for oc observe command

### LINKED ISSUE #22274: Add Label selector support to oc observe
I'd like to filter objects with Label selector in `oc observe` command.

```
oc observe services -a '{ .spec.clusterIP }' -l app=myapp -- register_dns.sh
```

### PR DESCRIPTION
**Pull request by superbrothers**
This PR adds label selector support for `oc observe` command.

```sh
# Observe changes to services filtered by a label selector	 
$ oc observe namespaces -l regist-dns=true -a '{ .spec.clusterIP }' -- register_dns.sh`)
```

/close #22274

### CODE REVIEW

File: pkg/oc/cli/observe/observe.go [hunk 1]
```diff
@@ -296,6 +302,10 @@ func (o *ObserveOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args [
 		return fmt.Errorf("you may only specify one argument containing the resource to observe (use '--' to separate your resource and your command)")
 	}
 
+	if _, err := labels.Parse(o.selector); err != nil {
```
Comment (soltysh) [inline pkg/oc/cli/observe/observe.go]: This belongs in `Validate` not `Complete`.

Comment (superbrothers) [reply pkg/oc/cli/observe/observe.go]: Fixed by eba61e7.

**Comment by superbrothers** [comment]
@soltysh Thanks for reviewing. I squashed changes.

**Pull request merged**